content/blog/migrating-vsphere-virtual-volumes-to-kubevirt-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,432 @@
+  csi.storage.k8s.io/controller-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/node-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-stage-secret-name: hpe-backend
+  csi.storage.k8s.io/node-stage-secret-namespace: hpe-storage
+  csi.storage.k8s.io/provisioner-secret-name: hpe-backend
+  csi.storage.k8s.io/provisioner-secret-namespace: hpe-storage
+  csi.storage.k8s.io/fstype: ext4
+  description: "Volume cloned or imported by the HPE CSI Driver for Kubernetes"
+  allowOverrides: importVolumeName,importVolAsClone
+  compression: "false"
+  provisioningType: ""
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+Users then call out the parameter and the value they desire in the PVC using an annotation.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: testdb-cloned
+  namespace: hpe-vmi
+  annotations:
+    csi.hpe.com/importVolAsClone: dat-mariadbl-56480669
+spec:
+  volumeMode: Block
+  storageClassName: legacy-migration
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 128Gi
+```
+
+So, where does “dat-mariadbl-56480669” come from? It’s the storage array volume name. It’s available to storage administrators of the storage array and it’s also available through the HPE Storage Integration Pack for VMware vCenter (SIP4VC) when used with HPE Alletra Storage MP B10000 and earlier platforms.
+
+![](/img/sip4vc.png "Virtual machine view in SIP4VC")
+
+Create the PVC and attach the PVC as a disk to the running KubeVirt VM.
+
+```shell
+$ kubectl create -f pvc-clone.yaml
+```
+
+Next, login to OKD.
+
+![](/img/attach-disk.png "Attach disk to VM")
+
+Once attached, a “migrate” playbook will mount the disk in inside the VM and fire up the database and test it.
+
+```yaml
+---
+# With PVC attached to the KubeVirt VM, mount and start MariaDB
+- hosts: kubevirt
+  become: yes
+  tasks:
+  - name: Mount filesystem and persist
+    mount:
+      path: "{{ var_lib_mysql }}"
+      state: mounted
+      src: "LABEL={{ filesystem_label }}"
+      fstype: ext4
+
+  - name: Ensure correct perms and SELinux labels
+    file:
+      path: "{{ var_lib_mysql }}"
+      owner: mysql
+      group: mysql
+      mode: "0755"
+      seuser: system_u
+      serole: object_r
+      setype: mysqld_db_t
+      selevel: s0 
+      recurse: yes
+  - name: Enable MariaDB
+    service:
+      name: mariadb
+      state: started
+      enabled: true
+
+  - name: Test Test_DB
+    shell: mysql -t < test_employees_sha.sql && mysql -t <<< 'select @@hostname as 'VM';'
+    args:
+      chdir: "{{ ansible_user_dir }}/test_db"
+    register: test_db_results
+
+  - name: Show results
+    pause:
+      prompt: "{{ test_db_results.stdout }}"
+      seconds: 10
+```
+
+Executing the playbook should reveal the test results and we can visually inspect the database is actually running on the KubeVirt VM.
```

Comment:
Modify the second half of the sentence to read "and **you** can **see** the database is actually running on the KubeVirt VM."